server/src/main/java/org/elasticsearch/index/IndexingPressure.java

```diff
@@ -79,6 +79,12 @@ public class IndexingPressure {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<ByteSizeValue> MAX_OPERATION_SIZE = Setting.memorySizeSetting(
+        "indexing_pressure.memory.max_operation_size",
+        "5%",
```

Comment:
This seems slightly breaking for small heaps, like 1GB heap now only allow 50MB docs. While this may be a good default, it could break some users too. Should we make it 10% here in the core code base for now?